yarn-project/simulator/src/avm/journal/nullifiers.ts

```diff
@@ -0,0 +1,158 @@
+import { Fr } from '@aztec/foundation/fields';
+
+import type { NullifiersDB } from '../../index.js';
+
+/**
+ * A class to manage new nullifier staging and existence checks during a contract call's AVM simulation.
+ * Maintains a pending nullifier cache, and ensures that existence checks fall back to the correct source.
+ * When a contract call completes, its pending nullifier set can be merged into its parent's.
+ */
+export class Nullifiers {
+  /** Cached nullifiers. */
```

Comment:
If its labeled Cached Nullifiers above should we just call the class that ?

Reply:
Good point